Automattic/simplenote-ios
# SearchResults: Specialized ViewController

### Details:
In this PR we're implementing **SearchResultsViewController**, a specialized VC in charge of rendering search results.

Since this is one of multiple PR's, in order to test this `Advanced Search` feature must be enabled.

@aerych I'm **VERY CURIOUS** about your thoughts on this one. I've done my best to... keep things simple + encapsulated, but we do have a number of custom things (inhouse SearchController, Blurred NavigationBar), and ... things require extra thought!!

Thank you in advance!!!

Ref. #507

### Test
1. Edit the [advancedSearchEnabled flag](https://github.com/Automattic/simplenote-ios/pull/527/files#diff-fdfb519fc818957652b45426448307d7R10) and set it to true
2. Log into your account
3. Tap over the **SearchBar**

- [x] Verify the new SearchResultsController shows up onscreen

### Release
These changes do not require release notes.
